xdsl/dialects/mod_arith.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC
+
+from xdsl.dialects.arith import signlessIntegerLike
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, operand_def, result_def
+from xdsl.traits import Pure
+
+
+class ModArithOp(IRDLOperation, ABC):
+    # assemblyFormat = "operands attr-dict `:` functional-type(operands, results)"
+    pass
+
+
+class BinaryOp(ModArithOp):
```

Comment:
Should this also inherit from `ABC`? (I'm not 100% sure how this works in python so I could be wrong

Reply:
I'm not fully sure, do I need to include it again when ModArithOp inherits from ABC?

Reply:
Yes, in general, every class that is "incomplete" (i.e. should not be instantiated itself but only serves to encapsulate common abstractions) should ideally be marked `ABC` to make it a type error when someone tries to instantiate them.